On newer kernels the arch dirs for x86 and x86_64 are merged into x86
but klibc does not know that. This patch teaches it where to look at
for the include files when KLIBCARCH=i386. Its already fixed when
KLIBCARCH=x86_64

-Khem

Index: klibc-1.5.18/usr/klibc/arch/i386/MCONFIG
===================================================================
--- klibc-1.5.18.orig/usr/klibc/arch/i386/MCONFIG	2010-07-01 15:36:11.175170001 -0700
+++ klibc-1.5.18/usr/klibc/arch/i386/MCONFIG	2010-07-01 15:36:57.875169996 -0700
@@ -26,6 +26,9 @@ KLIBCARCHREQFLAGS += $(gcc_m32_option)
 
 KLIBCBITSIZE  = 32
 
+# Asm includes for i3?6  are in the merged x86 tree
+KLIBCARCHINCFLAGS = -I$(KLIBCKERNELOBJ)/arch/x86/include
+
 # Extra linkflags when building the shared version of the library
 # This address needs to be reachable using normal inter-module
 # calls, and work on the memory models for this architecture
